packages/server/graphql/mutations/createGitHubTaskIntegration.ts

```diff
@@ -68,44 +71,77 @@ export default {
       })
     }
 
-    // RESOLUTION
     const {content: rawContentStr, meetingId} = task
-    const viewerAuth = await dataLoader.get('githubAuth').load({teamId, userId: viewerId})
-
-    if (!viewerAuth) {
+    const [viewerAuth, assigneeAuth, team] = await Promise.all([
+      dataLoader.get('githubAuth').load({teamId, userId: viewerId}),
+      userId && dataLoader.get('githubAuth').load({teamId, userId}),
+      dataLoader.get('teams').load(teamId)
+    ])
+    const auth = assigneeAuth ?? viewerAuth
+    if (!auth) {
       return standardError(
-        new Error(`Assignment failed! The assignee does not have access to GitHub`),
+        new Error(`Assignment failed! Neither you nor the assignee has access to GitHub`),
         {userId: viewerId}
       )
     }
+    const teamMembers = await dataLoader.get('teamMembersByTeamId').load(teamId)
+    const viewer = teamMembers.find(({userId}) => userId === viewerId)
+    const assignee = userId && teamMembers.find((user) => user.userId === userId)
+
+    if (!viewer || (userId && !assignee)) {
+      return standardError(new Error('Not part of the team'), {
+        userId: viewerId,
+        tags: {
+          userId: userId ?? '',
+          viewerId,
+          teamId,
+          taskId
+        }
+      })
+    }
+
+    // RESOLUTION
+    const accessUserId = assigneeAuth && userId ? userId : viewerId
```

Comment:
the person calling the mutation is the person who likely has the most access to jira (i know this logic doesn't apply to github, but if they change permissions we'll be ready. plus, it's just nice to have both integrations using the same logic chain)

. for example, maybe the assignee can't create tasks in jira?

let's favor using the viewer's auth first:

```ts
const accessUserId = viewerAuth ? viewerId : userId
```

Reply:
I used assignee first because then the assignee matches in GitHub/Jira. Can do it the other way round.